app/helpers/products_helper.rb

```diff
@@ -6,4 +6,36 @@ def format_quantity_adjustment(quantity)
 
     quantity.to_s
   end
+
+  def stock_warnings(product)
+    low_stock_sum = low_stock_count(product)
+    out_of_stock_sum = out_of_stock_count(product)
+
+    if low_stock_sum.positive? && out_of_stock_sum.positive?
+      "<span class='badge text-bg-warning'>Low stock in #{low_stock_sum} locations.</span><br>
+      <span class='badge text-bg-danger'>Out of stock in #{out_of_stock_sum} locations.</span>".html_safe
+    elsif out_of_stock_sum.positive?
+      "<span class='badge text-bg-danger'>Out of stock in #{out_of_stock_sum} locations.</span>".html_safe
+    elsif low_stock_sum.positive?
+      "<span class='badge text-bg-warning'>Low stock in #{low_stock_sum} locations.</span>".html.safe
+    end
```

Comment:
Maybe we could DRY it up?

I would do something like:

```
warnings = []
warnings << low_stock_warning(low_stock_sum) if low_stock_sum.positive?
warnings << out_of_stock_warning(out_of_stock_sum) if out_of_stock.positive?
warnings.join("<br>").html_safe
```

`low_stock_warning/1` and `out_of_stock_warning/1` are private functions that return the HTML string.

Reply:
We could also fix pluralization here :)